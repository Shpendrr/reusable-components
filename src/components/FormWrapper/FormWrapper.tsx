import React, { ReactNode } from "react";

interface IFormWrapperProps {
  children: ReactNode;
}

const FormWrapper: React.FC<IFormWrapperProps> = ({ children, ...rest }) => {
  return <form {...rest}>{children}</form>;
};

export default FormWrapper;
