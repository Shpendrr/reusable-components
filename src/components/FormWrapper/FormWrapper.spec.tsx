import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import FormWrapper from "@/components/FormWrapper/FormWrapper";

describe("FormWrapper", () => {
  it("renders children", () => {
    const { getByText } = render(
      <FormWrapper>
        <div>Child element</div>
      </FormWrapper>
    );
    expect(getByText("Child element")).toBeInTheDocument();
  });

  it("matches the snapshot", () => {
    const { container } = render(
      <FormWrapper>
        <div>Child element</div>
      </FormWrapper>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
