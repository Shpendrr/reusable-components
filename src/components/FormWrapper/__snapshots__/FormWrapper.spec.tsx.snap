// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormWrapper matches the snapshot 1`] = `
<form>
  <div>
    Child element
  </div>
</form>
`;
